### PR TITLE
Add mongoose schemas and enfore code style

### PR DESCRIPTION
**Pull request by Voakie**

### CODE REVIEW

**Review by yJulian — COMMENTED** [review]
Looks nice 🐱‍👤🧮😜✌

**Review by yJulian — APPROVED** [review]
I think we can approve it 🐱‍👤✌😜

File: src/models/unit.types.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  g = "g",
+  pcs = "pcs",
+  prt = "prt",
+  amt = "amt",
+  tsp = "tsp",
+  tbls = "tbls"
+}
+
+export interface IUnitSchema extends Document {
+  type: string | EIngredientUnit;
```
Comment (DeveloperTK) [inline src/models/unit.types.ts:14]: EIngredientUnit could be enforced with the model based mongoose

**Review by DeveloperTK — CHANGES_REQUESTED** [review]
Please add a salt (string) in the user table

File: src/models/user.schema.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { model, Schema, Types } from "mongoose";
+import { IUserSchema } from "./user.types";
+
+const userSchema = new Schema<IUserSchema>({
+  _id: Types.ObjectId,
+  created: Date,
+  email: String,
+  username: String,
+  password: String
```
Comment (DeveloperTK) [inline src/models/user.schema.ts:9]: please salt the passwords with individual salts per user

**Review by DeveloperTK — APPROVED** [review]
Nice. Keep up the good work!

**Comment by DeveloperTK** [comment]
Although I would have appreciated it if you left three lines out to get to 420 added lines...

File: src/models/user.schema.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { model, Schema, Types } from "mongoose";
+import { IUserSchema } from "./user.types";
+
+const userSchema = new Schema<IUserSchema>({
+  _id: Types.ObjectId,
+  created: Date,
+  email: String,
+  username: String,
+  password: String,
```
Comment (finnmglas) [inline src/models/user.schema.ts:9]: Obviously **not the actual password** but rather a hash... we should hash it on the clients device and only store/ send the hash to the server :lock:

**Review by finnmglas — APPROVED** [review]

**Comment by finnmglas** [comment]
I suppose we don't need 8 reviewers :laughing:

**Pull request merged**